public/javascripts/checkout.js

```diff
@@ -65,47 +64,50 @@ lichess.checkout = function(publicKey) {
     $checkout.find('.service').html(lichess.spinnerHtml);
   });
 
-  $checkout.find('button.stripe').on('click', function() {
-    var freq = getFreq(), usd, amount;
+  let stripe = Stripe(publicKey);
+  let showError = (error) => {
+    // TODO: make this show an actual error
+    console.log(error);
+  };
+  $checkout.find('button.stripe').on('click', function () {
+    var freq = getFreq(), amount;
     if (freq == 'lifetime') {
-      usd = lifetime.usd;
       amount = lifetime.cents;
     } else {
       var $input = $checkout.find('group.amount input:checked');
-      usd = $input.data('usd');
       amount = parseInt($input.data('amount'));
     }
     if (amount < min || amount > max) return;
-    $stripeForm.find('.amount').val(amount);
-    $stripeForm.find('.freq').val(freq);
-    var desc = freq === 'monthly' ? usd + '/month' : usd + ' one-time';
+    $checkout.find('.service').html(lichess.spinnerHtml);
 
-    stripeHandler.open({
-      description: desc,
-      amount: amount,
-      panelLabel: '{{amount}}',
-      email: $checkout.data('email')
-    });
+    $.ajax({
+      url: "/patron/stripe-checkout",
+      method: "post",
+      data: {
+        email: $checkout.data('email'),
+        amount: amount,
+        freq: freq
+      }
+    }).then(
+      data => {
+        if (data.session && data.session.id) {
+          stripe.redirectToCheckout({
+            sessionId: data.session.id
+          }).then(function (result) {
+            showError(result.error.message);
```

Comment:
why is there an error here? It looks like a success workflow

Reply:
You are correct that it looks like a success workflow. However,  My reading of the stripe documentation says this is what I'm supposed to do:

https://stripe.com/docs/payments/checkout/one-time#redirect-checkout I wouldn't mind a second opinion though.